### PR TITLE
Appsync API Cache Implementation

### PR DESCRIPTION
**Pull request by mattheidelbaugh**
Implementation for all Appsync API Cache APIs

This is my first PR so be brutal :)

### CODE REVIEW

File: moto/appsync/models.py [hunk 1]
```diff
@@ -132,6 +137,61 @@ def to_json(self) -> Dict[str, Any]:
...
+            self.transit_encryption_enabled = False
+        else:
+            self.transit_encryption_enabled = transit_encryption_enabled
+
+        if at_rest_encryption_enabled is None:
+            self.at_rest_encryption_enabled = False
+        else:
+            self.at_rest_encryption_enabled = at_rest_encryption_enabled
+
+        if health_metrics_config is None:
```
Comment (armichaud) [inline moto/appsync/models.py:164]: `self.health_metrics_config = health_metrics_config or "DISABLED"`

File: moto/appsync/responses.py [hunk 2]
```diff
@@ -192,3 +192,60 @@ def get_introspection_schema(self) -> str:
...
+    def delete_api_cache(self) -> str:
+        api_id = self.path.split("/")[-2]
+        self.appsync_backend.delete_api_cache(
+            api_id=api_id,
+        )
+        return "{}"
+
+    def create_api_cache(self) -> str:
+        params = json.loads(self.body)
+        api_id = self.path.split("/")[-2]
```
Comment (armichaud) [inline moto/appsync/responses.py:212]: I *think* you can get this var with `self._get_param("appId")`

File: moto/appsync/responses.py [hunk 3]
```diff
@@ -192,3 +192,60 @@ def get_introspection_schema(self) -> str:
...
+            transit_encryption_enabled=transit_encryption_enabled,
+            at_rest_encryption_enabled=at_rest_encryption_enabled,
+            api_caching_behavior=api_caching_behavior,
+            type=type,
+            health_metrics_config=health_metrics_config,
+        )
+        return json.dumps(dict(apiCache=api_cache.to_json()))
+
+    def update_api_cache(self) -> str:
+        api_id = self.path.split("/")[-3]
```
Comment (armichaud) [inline moto/appsync/responses.py:231]: See comment above

File: moto/appsync/responses.py [hunk 4]
```diff
@@ -192,3 +192,60 @@ def get_introspection_schema(self) -> str:
...
+            api_id=api_id,
+            ttl=ttl,
+            api_caching_behavior=api_caching_behavior,
+            type=type,
+            health_metrics_config=health_metrics_config,
+        )
+        return json.dumps(dict(apiCache=api_cache.to_json()))
+
+    def flush_api_cache(self) -> str:
+        api_id = self.path.split("/")[-2]
```
Comment (armichaud) [inline moto/appsync/responses.py:247]: Same

**Comment by armichaud** [comment]
@bblommers , @mattheidelbaugh is a teammate of mine. All of my comments are NITs

File: moto/appsync/responses.py (hunk 2 above)
Comment (mattheidelbaugh) [reply moto/appsync/responses.py:212]: Yeah I saw that function elsewhere, but it doesn't seem to work in this case.  Not quite sure why

**Pull request merged**